### PR TITLE
[New Rule] Rapid7 Threat Command CVEs Correlation

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/integrations/issues/7360

## Summary
This pull request adds the `Rapid7 Threat Command CVEs Correlation` threat intel rule. For more information, please refer to the related issue.

Note: We had to add the `ti_rapid7_threat_command` integration manifests and schemas since we are referencing this in the rule. Additionally, we slightly adjusted the `test_integration_tag()` unit test to ignore `threat_match` rule types when referring to integration mappings to selected indexes in the rule. The indexes selected may differ from the actual threat index.

Also, a rules folder called `threat_intel` has been made and all threat intel related rules have been moved into this folder rather than keep them in `cross-platform`. This is more subjective on being better organized, however, there are several other integration related threat_intel integrations and packages, thus as we build these out with respective rules they have a dedicated folder.

## Testing
`view-rule` has been used to check the validity of the rule. Additionally, a package has been built and used to import this rule in Kibana API format to ensure that it imports correctly.

Screenshot 2024-05-29 at 4 23 49 PM

[NDJSON file for importing rule into Kibana](https://github.com/elastic/detection-rules/files/15490430/8.15-consolidated-rules.ndjson.txt)

### CODE REVIEW

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (w0rk3r) [inline rules/threat_intel/threat_intel_rapid7_threat_command.toml:90]: It looks very specific, Does it match any of the provided indexes? Some of them are very high volume and could cause issues. Maybe we should include only vuln scan integrations?

Comment (brokensound77) [inline rules/threat_intel/threat_intel_rapid7_threat_command.toml:90]:
```suggestion
query = '''
vulnerability.id : *
'''
```

Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:90]: Great question - This is the original rule that is being ported over, it's been adjusted slightly based on my review of other threat intel rules.

https://github.com/elastic/integrations/blob/main/packages/ti_rapid7_threat_command/kibana/security_rule/eaecc8f0-6704-11ed-80b2-9bbc46f73b72.json

File: detection_rules/schemas/definitions.py
Comment (kcreddy) [inline detection_rules/schemas/definitions.py:44]: Does this addition make other detection rules such as `threat_intel_indicator_match*` not eligible to be run on `ti_rapid7_threat_command` integration? 

https://github.com/elastic/detection-rules/blob/2a8ea15fa4bf8e2e06e57c3cac547cdfba539d20/detection_rules/rule.py#L1220

We do want all the 4 indicator match rules named `threat_intel_indicator_match*` to be able to run for this integration.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [inline rules/threat_intel/threat_intel_rapid7_threat_command.toml:73]: Although the rule is related to TI integration, I think we are technically not matching the threat indicators, but the vulnerabilities. Maybe `Rule Type: Vulnerability Match` makes more sense if that tag allowed to exist.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [inline rules/threat_intel/threat_intel_rapid7_threat_command.toml:79]: The index is going to be different here. The PR: https://github.com/elastic/integrations/pull/9925 changes this destination index name. Going forwards, this value should be as per transform's index alias name i.e., `logs-ti_rapid7_threat_command_latest.ioc`.

https://github.com/elastic/integrations/blob/3d2e964f3897411a359f9dbc1030820b09f09bc8/packages/ti_rapid7_threat_command/elasticsearch/transform/latest_ioc/transform.yml#L19

Once this rule is available, we will update integration doc (before merging the above PR) to use this rule instead of the custom rule which was shipped along with the package in earlier versions.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [inline rules/threat_intel/threat_intel_rapid7_threat_command.toml:86]:
```suggestion
timeline_title = "Generic Vulnerability Match Timeline"
```

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml]: Well, I think the users could also ingest events from the respective beats and the indices in question does seem to be appropriate match `index = ["auditbeat-*", "endgame-*", "filebeat-*", "logs-*", "packetbeat-*", "winlogbeat-*"]`. It also matches with existing rule already. 

Maybe for the `logs-*`, it is a bit broad and can be narrowed down, but I don't have a vuln scan integrations list ready. Also, we don't have any naming conventions on vulnerability integrations like we do `ti_*` for Threat Intel integrations, so this rule should keep changing to include newer integrations as they are added and maintaining it could be an issue.

**Comment by kcreddy** [comment]
Hey @terrancedejesus, I addressed few comments here: https://github.com/elastic/integrations/issues/7360#issuecomment-2138798243. Let me know if there anything pending from my end to proceed further.

File: detection_rules/schemas/definitions.py
Comment (terrancedejesus) [reply detection_rules/schemas/definitions.py:44]: Nope. This addition allows for us to properly identify how to add `related_integrations` data to the prebuilt rule object.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:73]: For `Rule Type` we attempt to match any actual rule type in the SIEM. If I understand correctly, this is a threat match type rule as indicated in the example you sent. https://github.com/elastic/integrations/blob/dfc057e4908d28fbe9f676ee4e3e0bdbe67cd35b/packages/ti_rapid7_threat_command/kibana/security_rule/eaecc8f0-6704-11ed-80b2-9bbc46f73b72.json#L80C17-L80C30

If that is the case, we can change it to `Rule Type: Threat Match` and `Use Case: Vulnerability` should still be searchable.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:79]: roger, so new `threat_index` value is `logs-ti_rapid7_threat_command_latest.ioc`?

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:86]: What is the timeline ID associated with this?

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml]: Thank you for the clarification, if a new index pattern is necessary to be agnostic with another integration we can always come back and tune the rule accordingly.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml]: > roger, so new threat_index value is logs-ti_rapid7_threat_command_latest.ioc?

Yes, that is correct.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml]: I don't think there is an existing timeline for vulnerability matches. So, it makes sense to remove the timeline here.

Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml]: Sounds good, removed.

**Review by approksiu — COMMENTED** [review]
Left a few questions.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (terrancedejesus) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:18]: cc @kcreddy - good question.

**Comment by terrancedejesus** [comment]
@kcreddy -> Here is the final rule, unless more changes are required. Note that you can checkout this branch and run the following command to get the Kibana formatted rule object as well.

`python -m detection_rules view-rule /Users/tdejesus/code/src/detection-rules/rules/threat_intel/threat_intel_rapid7_threat_command.toml`

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:83]: > Does this integration support expiring indicators?

We are in the process of adding the support in https://github.com/elastic/integrations/pull/9925, which is going to use the existing built-in `Threat Intel` rules, i.e., `and not labels.is_ioc_transform_source:"true"` once that PR is merged. This PR rule is for vulnerabilities and not indicators, so we are good not having that.

> Not sure that vulnerabilities would ever expire

Yeah, we don't have support for expiring vulnerabilities.

File: rules/threat_intel/threat_intel_rapid7_threat_command.toml
Comment (kcreddy) [reply rules/threat_intel/threat_intel_rapid7_threat_command.toml:18]: Not quite sure what `high max_signals count rules` are meant for, but I think that if the [existing custom rule](https://github.com/elastic/integrations/blob/main/packages/ti_rapid7_threat_command/kibana/security_rule/eaecc8f0-6704-11ed-80b2-9bbc46f73b72.json) is not already in that list, then this rule is probably okay to be left out as well.

**Comment by terrancedejesus** [comment]
@approksiu @kcreddy - Happy to get this merged in, but need approvals. I believe all questions and suggestions have been addressed.

**Review by kcreddy — APPROVED** [review]
Thanks so much Terrance for adding this rule. LGTM 👍🏼

**Comment by approksiu** [comment]
@terrancedejesus @kcreddy I think we should set max_signals to 10000 maximum at the moment. We'll do more research on the implications of increasing that number. I assume with the previously limited visibility of max_signals and its setup for users, the original prebuilt rule was limited by a system default value.

Apart from this comment I am fine with the rule.

cc @yctercero

**Review by approksiu — APPROVED** [review]
Thanks for addressing my comment.

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Pull request merged**
